File: Frontend/src/app/models/dtos/LocalResponse.ts
import { UserResponse } from './UserResponse';
import { EventBasicInfo } from './EventBasicInfo';

//Respuesta detallada de un local, incluyendo el usuario propietario y sus eventos.
export interface LocalResponse {
  id: number;
  name: string;
  latitude: number;
  longitude: number;
  ubication: string;
  capacity: number;
  rooms: number;
  user: UserResponse;
  events: EventBasicInfo[];
}

File: Backend/daw2026/src/main/java/daw2026/Dto/LocalResponse.java
package daw2026.Dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class LocalResponse {
    
    private Long id;
    private String name;
    private int latitude;
    private int longitude;
    private String ubication;
    private int capacity;
    private int rooms;
    private UserResponse user;
    private List<EventBasicInfo> events;
}
